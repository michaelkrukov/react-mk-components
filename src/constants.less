// Colors
@primary: #5F1EC4;
@secondary: #E0E0E0;
@danger: #dc3545;
@success: #28a745;
@warning: #f8ac19;
@info: #17a2b8;
@dark: #343a40;
@light: #f8f9fa;

// Sizes
@sm: 576px;
@md: 768px;
@lg: 992px;
@xl: 1200px;

@c-sm: 540px;
@c-md: 720px;
@c-lg: 960px;
@c-xl: 1140px;
