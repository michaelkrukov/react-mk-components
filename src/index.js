export {
    default as Button
} from './components/Button';
export {
    default as Container
} from './components/Container';
export {
    default as Modal,
    ModalHeader,
    ModalBody,
    ModalFooter
} from './components/Modal';
export {
    default as Transition
} from './components/Transition';
export {
    default as BodyEnd
} from './components/BodyEnd';
