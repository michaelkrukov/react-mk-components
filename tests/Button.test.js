import React from 'react';
import { Button } from '../src';
import renderer from 'react-test-renderer';

test('Button renders', () => {
  const component = renderer.create(
    <Button>button</Button>,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button renders with color', () => {
  const component = renderer.create(
    <Button primary>button</Button>,
    <Button danger>button</Button>,
    <Button success>button</Button>,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button renders as block', () => {
  const component = renderer.create(
    <Button block>button</Button>,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
