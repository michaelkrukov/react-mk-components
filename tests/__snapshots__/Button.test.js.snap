// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders 1`] = `
<button
  className="Button"
  type="button"
>
  button
</button>
`;

exports[`Button renders as block 1`] = `
<button
  className="Button block"
  type="button"
>
  button
</button>
`;

exports[`Button renders with color 1`] = `
<button
  className="Button primary"
  type="button"
>
  button
</button>
`;
